apps/src/aichat/views/modelCustomization/MultiInputCustomization.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {useState, useCallback} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import {StrongText} from '@cdo/apps/componentLibrary/typography/TypographyElements';
+
+import {Visibility} from '../../types';
+
+import FieldLabel from './FieldLabel';
+import {isDisabled} from './utils';
+
+import styles from './retrieval-customization.module.scss';
+import modelCustomizationStyles from '../model-customization-workspace.module.scss';
+
+const MultiInputCustomization: React.FunctionComponent<{
+  label: string;
+  fieldId: string;
+  tooltipText: string;
+  addedItems: string[];
+  visibility: Visibility;
+  isReadOnly: boolean;
+  hideInputsWhenReadOnly: boolean;
+  onUpdateItems: (updatedItems: string[]) => void;
+}> = ({
+  label,
+  fieldId,
+  tooltipText,
+  addedItems,
+  visibility,
+  isReadOnly,
+  hideInputsWhenReadOnly,
+  onUpdateItems,
+}) => {
+  const [newItem, setNewItem] = useState('');
+
+  const onRemove = useCallback(
+    (index: number) => {
+      const newItems = [...addedItems];
+      newItems.splice(index, 1);
+      onUpdateItems(newItems);
+    },
+    [addedItems, onUpdateItems]
+  );
+
+  const onAdd = useCallback(() => {
+    onUpdateItems([newItem, ...addedItems]);
+    setNewItem('');
+    document.getElementById(fieldId)?.focus();
+  }, [newItem, addedItems, fieldId, onUpdateItems]);
+
+  return (
+    <>
+      {(!isReadOnly || !hideInputsWhenReadOnly) && (
+        <>
+          <div className={modelCustomizationStyles.inputContainer}>
+            <FieldLabel label={label} id={fieldId} tooltipText={tooltipText} />
+            <textarea
+              id={fieldId}
+              onChange={event => setNewItem(event.target.value)}
+              value={newItem}
+              disabled={isReadOnly}
+            />
+          </div>
+          <div className={styles.addItemContainer}>
+            <Button
+              text="Add"
+              type="secondary"
+              color="gray"
+              size="s"
+              onClick={onAdd}
+              iconLeft={{iconName: 'plus'}}
+              disabled={!newItem || isReadOnly}
+            />
+          </div>
+        </>
+      )}
+      <div className={styles.addedItemsHeaderContainer}>
+        <StrongText>Added</StrongText>
+      </div>
+      {addedItems.map((message, index) => {
+        return (
+          <div key={index} className={styles.itemContainer}>
```

Comment:
What do you think of increasing `padding`'s 2nd value in `itemContainer`?
Current at 12px:
![Screenshot 2024-08-29 at 2 25 24 PM](https://github.com/user-attachments/assets/15b409c2-59c0-42d2-933b-400778ab5c5a)

Reply:
If I were to change the padding on these, I'd probably want to decrease the padding to match up better with the figma design - 

![alt text](image.png)

https://www.figma.com/design/tjkMqlAXi2iTqpUHSljB3A/Gen-AI-Unit?node-id=2553-18651&t=5gG1Rndcnorr0SlS-4

Reply:
If I were to change the padding on these, I'd probably want to decrease the padding to match up better with the figma design - 

![image](https://github.com/user-attachments/assets/70f4526a-5029-4340-bc60-b4f4b80ce449)


https://www.figma.com/design/tjkMqlAXi2iTqpUHSljB3A/Gen-AI-Unit?node-id=2553-18651&t=5gG1Rndcnorr0SlS-4

Reply:
Oh I misunderstood that as itemContainer margin. For the padding, do you mean so that the `x` icon and the text aren't so close together?

Reply:
Yes 😄

Reply:
Yeeeaaah that's heckin jank. I think the `4px 12px` padding should really be on itemContainer > span? (can you do that on inline elements?) The `x` is also a bit off from the design (centered instead of top right). 

![image](https://github.com/user-attachments/assets/f620c5cf-3a8a-4135-98cc-134385258961)

Let me fiddle with the css a bit and see if I can make that better.

Reply:
Actually, are you OK with me including those style changes in the next PR? It'll be a little more scoped and changing the same area of the UI. That way I don't have to re-do all the screenshots/videos for this one :P

Reply:
Oh yeah - that sounds good to me!